plugins/com.gwtplugins.gwt.eclipse.core/src/com/google/gwt/eclipse/core/dialogs/ModuleSelectionDialog.java

```diff
@@ -1,499 +1,497 @@
+      IModule module = (IModule) result[0];
+      return module;
+    }
+
+    return null;
+  }
+
+  /**
+   * Helper method to return the absolute workspace path of a GWT Module.
+   *
+   * If the module file is located in a JAR, then the absolute path of the JAR
+   * on the file system is returned.
+   */
+  private static IPath getPathForModule(IModule module) {
+
+    if (module instanceof ModuleFile) {
+      ModuleFile moduleFile = (ModuleFile) module;
+      IFile file = moduleFile.getFile();
+      return file.getFullPath();
+    }
+    else if(module instanceof ModuleJarResource) {
+      ModuleJarResource moduleJarResource = (ModuleJarResource) module;
+      IJarEntryResource jarEntryResource = moduleJarResource.getJarEntryResource();
+      return jarEntryResource.getPackageFragmentRoot().getPath();
+    }
+    else {
+      return null;
+    }
+  }
+
+  private IJavaProject javaProject;
+  private boolean showModulesInJars;
+
+  private ModuleSelectionDialog(Shell shell, IJavaProject javaProject,
+      boolean showModulesInJars) {
+    super(shell);
+    this.javaProject = javaProject;
+    this.showModulesInJars = showModulesInJars;
+    setTitle("GWT Module Selection");
+    setListLabelProvider(new ModuleItemLabelProvider());
+    setDetailsLabelProvider(new ModuleItemDetailsLabelProvider());
+  }
+
+  @Override
+  public String getElementName(Object item) {
+    IModule module = (IModule) item;
+    return module.getSimpleName();
+  }
+
+  @Override
+  protected Control createExtendedContentArea(Composite parent) {
+    // No need for an extended content area.
+    return null;
+  }
+
+  @Override
+  protected ItemsFilter createFilter() {
+    return new ModuleItemsFilter();
+  }
+
+  @Override
+  protected void fillContentProvider(AbstractContentProvider contentProvider,
+      ItemsFilter itemsFilter, IProgressMonitor progressMonitor) {
+
+    /*
+     * TODO: Should probably try and use the progressMonitor appropriately here.
+     * If we end up searching the whole workspace for modules, this could take a
+     * while, and we need some accurate indication of progress.
+     */
+
+    IModule[] projectModules = ModuleUtils.findAllModules(javaProject,
+        showModulesInJars);
+
+    for (IModule module : projectModules) {
+      contentProvider.add(module, itemsFilter);
+    }
+  }
+
+  @Override
+  protected IDialogSettings getDialogSettings() {
+    IDialogSettings settings = IDEWorkbenchPlugin.getDefault().getDialogSettings().getSection(
+        DIALOG_SETTINGS);
+
+    if (settings == null) {
+      settings = IDEWorkbenchPlugin.getDefault().getDialogSettings().addNewSection(
+          DIALOG_SETTINGS);
+    }
+    return settings;
+  }
+
+  @Override
+  protected Comparator<?> getItemsComparator() {
+    return (Object o1, Object o2) -> {
+      Collator collator = Collator.getInstance();
+      IModule module1 = (IModule) o1;
+      IModule module2 = (IModule) o2;

```

Comment:
While you're at it, change this to specify the right type, so a cast isn't even needed?
```suggestion
  @Override
  protected Comparator<IModule> getItemsComparator() {
    return (Object module1, Object module2) -> {
      Collator collator = Collator.getInstance();

```

Reply:
Ok!